lib/screens/calendar/calendar.dart

```diff
@@ -0,0 +1,841 @@
+import 'dart:async';
+import 'dart:collection';
+import 'package:brunstadtv_app/graphql/client.dart';
+import 'package:brunstadtv_app/graphql/queries/calendar.graphql.dart';
+import 'package:brunstadtv_app/router/router.gr.dart';
+import 'package:flutter_riverpod/src/consumer.dart'
+    show ConsumerState, ConsumerStatefulWidget;
+import 'package:flutter/material.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:riverpod/riverpod.dart';
+import 'package:table_calendar/table_calendar.dart';
+import 'package:intl/intl.dart';
+
+class CalendarPage extends ConsumerStatefulWidget {
+  const CalendarPage({super.key});
+
+  @override
+  ConsumerState<CalendarPage> createState() => _CalendarPageState();
+}
+
+class _CalendarPageState extends ConsumerState<CalendarPage> {
+  DateTime _focusedDay = DateTime.now();
+  DateTime? _selectedDay = DateTime.now();
+
+  bool _isCalendarCollapse = true;
+
+  Future<Query$CalendarPeriod$calendar$period?>? _calendarPeriod;
+  Future<Query$CalendarDay$calendar?>? _selectedDayEvent = null;
+
+  HashSet<DateTime> activeDaysPeriod = HashSet<DateTime>();
+  List<String> eventsPeriod = [];
+
+  final kFirstDay = DateTime(
+      DateTime.now().year - 2, DateTime.now().month, DateTime.now().day);
+  final kLastDay = DateTime(
+      DateTime.now().year + 2, DateTime.now().month, DateTime.now().day);
+
+  List<DateTime> _getEventsForDay(DateTime day) {
+    List<DateTime> result = [];
+    if (activeDaysPeriod != null) {
+      for (var i in activeDaysPeriod) {
+        if (i.year == day.year && i.month == day.month && i.day == day.day) {
+          result.add(i);
+        }
+      }
+      return result;
+    } else
+      return List.filled(0, DateTime.now());
+  }
+
+  bool _isSameWeekAsCurrentWeek(DateTime date) {
+    return _getWeekNumber(date).toString() ==
+        _getWeekNumber(DateTime.now()).toString();
+  }
+
+  /// Calculates number of weeks for a given year as per https://en.wikipedia.org/wiki/ISO_week_date#Weeks_per_year
+  int numOfWeeks(int year) {
+    DateTime dec28 = DateTime(year, 12, 28);
+    int dayOfDec28 = int.parse(DateFormat("D").format(dec28));
+    return ((dayOfDec28 - dec28.weekday + 10) / 7).floor();
+  }
+
+  /// Calculates week number from a date as per https://en.wikipedia.org/wiki/ISO_week_date#Calculation
+  int _getWeekNumber(DateTime date) {
+    int dayOfYear = int.parse(DateFormat("D").format(date));
+    int woy = ((dayOfYear - date.weekday + 10) / 7).floor();
+    if (woy < 1) {
+      woy = numOfWeeks(date.year - 1);
+    } else if (woy > numOfWeeks(date.year)) {
+      woy = 1;
+    }
+    return woy;
+  }
+
+  String convertToIso8601(DateTime time) {
+    //need to have a manual convert because the default method have a special 's' -> :sssZ
+    return DateFormat("yyyy-MM-ddTHH:mm:ss'Z'").format(time);
+  }
+
+  String convertToIso8601inDays(DateTime time) {
+    String timeIn8601 = DateFormat("yyyy-MM-ddTHH:mm:ss'Z'").format(time);
```

Comment:
you can modify the DateFormat() here to just be the first 10 characters instead of using substring.
```suggestion
    String timeIn8601 = DateFormat("yyyy-MM-dd").format(time);
```